subsys/net/lib/sockets/sockets_inet.c

```diff
@@ -1973,6 +1973,18 @@ int zsock_getsockopt_ctx(struct net_context *ctx, int level, int optname,
 			}
 
 			break;
+#if defined(CONFIG_NET_IPV4)
+		case IP_MULTICAST_LOOP:
+			if (*optlen != sizeof(int)) {
+				errno = EINVAL;
+				return -1;
+			}
+
+			*(int *)optval = ctx->options.ipv4_mcast_loop ? 1 : 0;
+			*optlen = sizeof(int);
+
+			return 0;
```

Comment:
Similar to #87338 I'd prefer to add a `NET_OPT_IPV4_MCAST_LOOP` to `enum net_context_option` and work with getter/setters here.